### ISSUE TITLE
[postgres] aggregate DDL: unparseable section

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3556]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Receiving parsing errors while linting aggregate-related PostgreSQL DDL.

### Expected Behaviour

No parsing errors.

### Observed Behaviour

```
==== sqlfluff ====
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
==== readout ====

=== [ path: 2022-07-05.aggregate.sql ] ===

== [2022-07-05.aggregate.sql] FAIL                                                                                                                                                
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | AGGREGATE range_agg_preserve_gaps...'
L:   8 | P:   1 |  PRS | Line 8, Position 1: Found unparsable section: 'CREATE OR
                       | REPLACE AGGREGATE range_agg_pr...'
L:  15 | P:   1 |  PRS | Line 15, Position 1: Found unparsable section: 'ALTER
                       | AGGREGATE range_agg_preserve_gaps ...'
L:  17 | P:   1 |  PRS | Line 17, Position 1: Found unparsable section: 'ALTER
                       | AGGREGATE my_agg (TSTZRANGE) OWNER...'
L:  18 | P:   1 |  PRS | Line 18, Position 1: Found unparsable section: 'ALTER
                       | AGGREGATE my_agg (TSTZRANGE) OWNER...'
L:  19 | P:   1 |  PRS | Line 19, Position 1: Found unparsable section: 'ALTER
                       | AGGREGATE my_agg (TSTZRANGE) OWNER...'
L:  20 | P:   1 |  PRS | Line 20, Position 1: Found unparsable section: 'ALTER
                       | AGGREGATE my_agg (TSTZRANGE) OWNER...'
L:  22 | P:   1 |  PRS | Line 22, Position 1: Found unparsable section: 'ALTER
                       | AGGREGATE my_agg (*) SET SCHEMA ap...'
L:  24 | P:   1 |  PRS | Line 24, Position 1: Found unparsable section: 'DROP
                       | AGGREGATE IF EXISTS my_agg (*) CASC...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
==== summary ====
violations:        9 status:         FAIL
All Finished!
```

### How to reproduce

Reproducible as above with the following file `2022-07-05.aggregate.sql`:
```sql
CREATE AGGREGATE range_agg_preserve_gaps (TSTZRANGE)
(
    sfunc = tstzrange_arr_merge,
    stype = TSTZRANGE[],
    initcond = '{}'
);

CREATE OR REPLACE AGGREGATE range_agg_preserve_gaps (TSTZRANGE)
(
    sfunc = tstzrange_arr_merge,
    stype = TSTZRANGE[],
    initcond = '{}'
);

ALTER AGGREGATE range_agg_preserve_gaps (TSTZRANGE) RENAME TO my_agg;

ALTER AGGREGATE my_agg (TSTZRANGE) OWNER TO me;
ALTER AGGREGATE my_agg (TSTZRANGE) OWNER TO CURRENT_ROLE;
ALTER AGGREGATE my_agg (TSTZRANGE) OWNER TO CURRENT_USER;
ALTER AGGREGATE my_agg (TSTZRANGE) OWNER TO SESSION_USER;

ALTER AGGREGATE my_agg (*) SET SCHEMA api;

DROP AGGREGATE IF EXISTS my_agg (*) CASCADE;
```

### Dialect

postgres

### Version

```
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
```

### Configuration

defaults

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by WittierDinosaur** [comment]
quite a hefty amount of syntax, linking the docs here https://www.postgresql.org/docs/14/sql-createaggregate.html

**Comment by R3gardless** [comment]
@alanmcruickshank Hi. Actually this is my second issue, but if possible, would it be okay for me to be assigned this and work on it?

**Comment by alanmcruickshank** [comment]
> @alanmcruickshank Hi. Actually this is my second issue, but if possible, would it be okay for me to be assigned this and work on it?

Done 👍 

Thanks for picking this up 🙏
